File: Java/프로그래머스/1/140108. 문자열 나누기/문자열 나누기.java
class Solution {
    public int solution(String s) {
        int answer = 0;
        char firstChar = '1'; 
        int firstCharCount = 0;
        int notFirstCharCount = 0;
        
        for(char c : s.toCharArray()) {
            if(firstCharCount == 0) {
                firstChar = c;
                answer++;
            }
            if(c == firstChar) {
                firstCharCount++;
            } else {
                notFirstCharCount++;
                if((firstCharCount == notFirstCharCount)) {
                    firstCharCount = 0;
                    notFirstCharCount = 0;
                    
                }
            }
        }
        return answer;
    }
}

File: Kotlin/프로그래머스/1/140108. 문자열 나누기/문자열 나누기.kt
class Solution {
    fun solution(s: String): Int {
        var answer: Int = 0
        var same = 0
        var diff = 0
        var firstChar : Char? = null
        for(i in 0..s.lastIndex) {
            if(firstChar==null) {
                firstChar = s[i]
                same ++
            } else if(firstChar == s[i]) {
                same++
            } else {
                diff++
            }
            if(same==diff || i==s.lastIndex) {
                same=0
                diff=0
                firstChar=null
                answer++
            }
        }
        return answer
    } 
}